src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java

```diff
@@ -136,31 +138,44 @@ public static boolean shouldAnalyzeAutomatically(@Nullable VirtualFile file, @Nu
     }
 
     // file and module not null here
-    if ("java".equalsIgnoreCase(file.getFileType().getDefaultExtension()) && !isSource(file, module)) {
-      SonarLintConsole.get(module.getProject()).debug("Not automatically analysing java file outside source folder: " + file.getName());
-      return false;
-    }
-
-    return true;
-  }
-
-  public static boolean isSource(VirtualFile file, Module module) {
     ModuleRootManager moduleRootManager = ModuleRootManager.getInstance(module);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Parameter 1 to this call is marked "org.jetbrains.annotations.NotNull" but null is passed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2637)